tests/page-objects/lending-bar-auto-renew.ts

```diff
@@ -0,0 +1,307 @@
+import { type Page, Locator, expect } from '@playwright/test';
+
+export class LendingBarAutoRenew {
+  readonly page: Page;
+
+  readonly iaBookActions: Locator;
+  readonly demoControls: Locator;
+
+  readonly loanSecondsFor5Min: Number;
+  readonly loanSecondsFor10Min: Number;
+  readonly loanSecondsFor30Min: Number;
+  readonly loanSecondsFor60Min: Number;
+
+  private timeInBrowser: Date;
+
+  public constructor(page: Page) {
+    this.page = page;
+
+    this.iaBookActions = this.page.locator('ia-book-actions');
+    this.demoControls = this.page.locator('.initial');
+
+    this.loanSecondsFor5Min = 300; // seconds
+    this.loanSecondsFor10Min = 600; // seconds 
+    this.loanSecondsFor30Min = 1800; // seconds
+    this.loanSecondsFor60Min = 3600; // seconds
+
+    this.timeInBrowser = new Date('2024-07-08T11:00:00'); // time
+  }
+
+  async clickOnBrowsedButton() {
+    const browsedCheckbox = this.demoControls.getByText('user_has_browsed', { exact: true });
+    await browsedCheckbox.click();
+  }
+
+  async gotoPage(uri: string) {
+    await this.page.goto(uri, { waitUntil: 'networkidle' });
```

Comment:
QUESTION/SUGGESTION: Shouldn't we be waiting for the appearance of something concrete, rather than for the network to be idle?